client/components/navigation-header/navigation-header.tsx

```diff
@@ -24,18 +25,33 @@ interface HeaderProps extends React.HTMLAttributes< HTMLElement > {
  * @param props - Component props
  * @param props.className - Additional CSS class name for the component
  * @param props.title - Header title text
+ * @param props.titleLogo - Header title logo element to render
  * @param props.backLinkProps - Object containing back link properties (backLink URL, backLinkText, and onBackClick handler)
  * @param props.titleElement - Custom element to override default title rendering
  * @param props.headElement - Custom element to override default head section rendering
- * @param props.children - Child elements to render in the right section
+ * @param props.rightSection - Child elements to render in the right section
  * @param props.hasScreenOptionsTab - Indicates whether the screen options tab should be added
  * @returns The rendered NavigationHeader component
  */
 const NavigationHeader: React.FC< HeaderProps > = ( {
 	className,
 	title,
+	titleLogo,
 	backLinkProps,
-	titleElement = <h1 className="calypso-navigation-header__title">{ title }</h1>,
+	titleElement = (
+		<h1 className="calypso-navigation-header__title">
+			{ titleLogo && (
```

Comment:
I would not expect the complex default prop to reduce the readability in the argument section. Could we move this initialization inside the content block?

Reply:
Agreed, moved this in the content block and also allowed this section to be overridden with a prop